src/lib/greenboot.rs

```diff
@@ -191,13 +208,102 @@ mod test {
     }
 
     #[test]
-    fn test_failed_diagnostics() {
-        setup_folder_structure(false)
-            .context("Test setup failed")
+    fn test_required_script_failure_exit_early() {
+        init_logger();
+        setup_folder_structure_for_test_required_script_failure_counter().expect("setup failed");
+
+        let base_path = GREENBOOT_INSTALL_PATHS[1];
+        let counter_file = format!("{}/fail_counter.txt", base_path);
+
+        let result = run_diagnostics(vec![]);
+        log::debug!("Diagnostics result: {:?}", result);
+
+        assert!(result.is_err());
+        assert_eq!(
+            result.unwrap_err().to_string(),
+            "health-check failed, skipping remaining scripts"
+        );
+
+        log::info!("Health check failed as expected.");
+
+        let fail_script_count = fs::read_to_string(counter_file)
+            .unwrap()
+            .trim()
+            .parse::<u32>()
             .unwrap();
-        let failed_msg = run_diagnostics(vec![]).unwrap_err().to_string();
-        assert_eq!(failed_msg, String::from("health-check failed!"));
-        tear_down().context("Test teardown failed").unwrap();
+        assert_eq!(
+            fail_script_count, 1,
+            "Only one failing script should have executed"
+        );
+
+        tear_down_setup_exit_early().expect("teardown failed");
+    }
+
+    fn setup_folder_structure_for_test_required_script_failure_counter() -> Result<()> {
```

Comment:
Extend the function `setup_folder_structure` to include multiple failing tests instead of writing new setup function, and add the counter in the test_.

You will also modify the  test_skip_failing_script to add newly added failing script in the skipped list.
here:
```
// Skip the failing script in required.d
let state = run_diagnostics(vec!["failing_script.sh".to_string()]);
```

Reply:
> Also as I understand from code test_skip_failing_script this test should be renamed to test_skip_disabled_script

Makes sense. Can you squeeze this into the current PR?

Reply:
sounds good, will make those changes and update the PR